p2p/transport.go

```diff
@@ -231,6 +231,13 @@ func (mt *MultiplexTransport) Listen(addr NetAddress) error {
 	return nil
 }
 
+// Returns the actual address to which the transport's listener is bound.
+// Useful when the port has been dynamically assigned by the OS, e.g. tcp://localhost:0
+func (mt *MultiplexTransport) ListenAddress() string {
```

Comment:
For purposes of `NodeInfo` we don't need to add this to the `Transport` interface, but we could...